kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java

```diff
@@ -38,83 +42,112 @@
-
-  @Test
-  public void createSubjectAccessReviewTest() {
-
-
-    KubernetesClient client = server.getClient();
-
-    SubjectAccessReview review = new SubjectAccessReviewBuilder().withNewSpec()
-      .withUser("admin-user")
-      .withNewResourceAttributes()
-      .withResource("pod")
-      .withVerb("create")
-      .endResourceAttributes()
-      .endSpec()
-      .build();
-
-    server.expect().post().withPath("/apis/authorization.k8s.io/v1/subjectaccessreviews").andReply(200, recordedRequest -> {
-      ObjectMapper mapper = new ObjectMapper();
-      SubjectAccessReview reviewResponse = null;
-      try {
-        reviewResponse = mapper.readValue(recordedRequest.getBody().readString(Charset.defaultCharset()), SubjectAccessReview.class);
-
-        reviewResponse.setStatus(new SubjectAccessReviewStatus(true, false, "",""));
-      } catch (IOException e) {
-        logger.error("Invalid response received", e);
-      }
-
-      return reviewResponse;
-    }).once();
-
-    SubjectAccessReview reviewResponse = client.subjectAccessReviewAuth().create(review);
-
-    assertNotNull(reviewResponse);
-    assertEquals(true, reviewResponse.getStatus().getAllowed());
-
-  }
-
-  @Test
-  public void createLocalSubjectAccessReviewTest() {
-
-
-    KubernetesClient client = server.getClient();
-
-    LocalSubjectAccessReview review = new LocalSubjectAccessReviewBuilder().withNewSpec()
-      .withUser("admin-user")
-      .withNewResourceAttributes()
-      .withResource("pod")
-      .withVerb("create")
-      .endResourceAttributes()
-      .endSpec()
-      .build();
-
-    server.expect().post().withPath("/apis/authorization.k8s.io/v1/namespaces/test/localsubjectaccessreviews").andReply(200, recordedRequest -> {
-      ObjectMapper mapper = new ObjectMapper();
-      LocalSubjectAccessReview reviewResponse = null;
-      try {
-        reviewResponse = mapper.readValue(recordedRequest.getBody().readString(Charset.defaultCharset()), LocalSubjectAccessReview.class);
-        reviewResponse.setStatus(new SubjectAccessReviewStatus(true, false, "",""));
-      } catch (IOException e) {
-        logger.error("Invalid response received", e);
-      }
-
-      return reviewResponse;
-    }).once();
-
-    LocalSubjectAccessReview reviewResponse = client.subjectAccessReviewAuth()
-      .inNamespace("test").create(review);
-
-    assertNotNull(reviewResponse);
-    assertNotNull(reviewResponse.getMetadata());
-    assertEquals("test", reviewResponse.getMetadata().getNamespace());
-    assertEquals("test", reviewResponse.getSpec().getResourceAttributes().getNamespace());
-    assertEquals(true, reviewResponse.getStatus().getAllowed());
-
-  }
-}
+    private static final Logger logger = LoggerFactory.getLogger(ClusterRoleBindingCrudTest.class);
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    public void createSubjectRulesReviewTest() {
```

Comment:
This test is testing 2 different scenarios, creating a self subject rule review programmatically (using DSL), and via deserialization. You should separate the test into 2 different tests (methods), one for each scenario.